MdeModulePkg/Library/VarCheckHiiLib/VarCheckHiiLib.inf

```diff
@@ -15,10 +15,10 @@
   MODULE_TYPE                    = DXE_RUNTIME_DRIVER
   VERSION_STRING                 = 1.0
   LIBRARY_CLASS                  = NULL|DXE_RUNTIME_DRIVER DXE_SMM_DRIVER
-  CONSTRUCTOR                    = VarCheckHiiLibNullClassConstructor
+  CONSTRUCTOR                    = VarCheckHiiLibSmmConstructor

```

Comment:
> Here, I don't think it makes sense to rename the name with smm. Because, this library instance is for runtime and smm both. I suggest to just remove smm from the function name and file name.

Renamed the file to VarCheckHiiLibSmm to clearly indicate its purpose for traditional SMM, but I agree that  include smm might not be appropriate since the library instance is intended for both runtime and SMM. I will push another version. Thanks